src/System.Runtime.Environment/src/OSName.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Runtime.InteropServices
+{
+    public struct OSName : IEquatable<OSName>
+    {
+        private string _name;
+
+        private const string WindowsName = "windows";
+        private const string LinuxName = "linux";
+        private const string OSXName = "osx";
+
+        private static readonly OSName s_windows = new OSName(WindowsName);
+        private static readonly OSName s_linux = new OSName(LinuxName);
+        private static readonly OSName s_osx = new OSName(OSXName);
+
+        public OSName(string name)
+        {
+            if (name.Equals("unix", StringComparison.OrdinalIgnoreCase))
+                _name = LinuxName;
+            else if (name.Equals("mac", StringComparison.OrdinalIgnoreCase) ||
+                name.Equals("macosx", StringComparison.OrdinalIgnoreCase))
+                _name = OSXName;
+            else _name = name;
+        }
+
+        public static OSName Windows { get { return s_windows; } }
+        public static OSName Linux { get { return s_linux; } }
+        public static OSName OSX { get { return s_osx; } }
+
+        public bool Equals(OSName osName)
+        {
+            if (osName._name.Equals(_name, StringComparison.OrdinalIgnoreCase))
+                return true;
+            return false;
+        }
```

Comment:
Should we override `GetHashCode` and `Equals(object)` as well?  Right now they are using strict equality semantics where the type uses ignore case semantics.